domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1894,12 +1899,28 @@ public TotaledResults(RowSet<Row> set, Integer estimatedTotal) {
                             Handler<AsyncResult<PostgresClientStreamResult<T>>> replyHandler) {
     // Start a transaction that we need to close.
     // If a transaction is already running we don't need to close it.
-    final Transaction transaction = connection.tx == null ? connection.conn.begin() : null;
+    if (connection.tx == null) {
+      connection.conn.begin()
+      .onFailure(cause -> {
+        log.error(cause.getMessage(), cause);
+        replyHandler.handle(Future.failedFuture(cause));
+      }).onSuccess(trans ->
+        executeGetQuery(connection, queryHelper, resultInfo, clazz, replyHandler, trans)
+      );
+    } else {
+      executeGetQuery(connection, queryHelper, resultInfo, clazz, replyHandler, null);
+    }
+  }
+
+  private <T> void executeGetQuery(SQLConnection connection, QueryHelper queryHelper,
+      ResultInfo resultInfo, Class<T> clazz,
+      Handler<AsyncResult<PostgresClientStreamResult<T>>> replyHandler, Transaction transaction) {
     connection.conn.prepare(queryHelper.selectQuery, prepareRes -> {
       if (prepareRes.failed()) {
-        closeIfNonNull(transaction);
+        closeIfNonNull(transaction).onComplete((voidRes) -> {
```

Comment:
```suggestion
        closeIfNonNull(transaction).onComplete(ignore -> {
```
Braces not needed.
`ignore` makes clear that we ignore a close failure on purpose.